lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -40,16 +40,16 @@
 )
 
 AZURE_CREDENTIAL_ENV_MAPPING = dict(
-    profile='AZURE_PROFILE',
-    subscription_id='AZURE_SUBSCRIPTION_ID',
-    client_id='AZURE_CLIENT_ID',
-    secret='AZURE_SECRET',
-    tenant='AZURE_TENANT',
-    ad_user='AZURE_AD_USER',
-    password='AZURE_PASSWORD',
-    cloud_environment='AZURE_CLOUD_ENVIRONMENT',
-    cert_validation_mode='AZURE_CERT_VALIDATION_MODE',
-    adfs_authority_url='AZURE_ADFS_AUTHORITY_URL'
+    profile=['AZURE_PROFILE', 'AZURE_PROFILE_ID', 'AZURE_CLIENT_PROFILE'],
```

Comment:
I do not think this change is necessary. There are no standards for this so it will never be compatible with all the tools.

Reply:
I agree with your premise, but I think if we were to pick one CI/CD tool to go out of our way to make Ansible compatible with, perhaps Jenkins should be that tool.  Also, this doesn't just make it work with one tool, but rather any permutations of env vars that may be available across several tools.  (I included all the Azure env var aliases that I observed in several contexts.)  The experience I want to have with Ansible is that "it just works" and I see this moving toward that goal, and I struggle to imagine plausible downsides.